### ISSUE TITLE
Scheduling update: causes infinite loop

### ISSUE THREAD

**Issue by Birkemose** [cocos2d/cocos2d-objc#514]
If user accidentally schedules update:, it will cause the app to enter an infinite update loop.

**Comment by slembcke** [comment]
Couldn't reproduce the infinite loop, but I am adding assertions for accidentally scheduling update methods.

**Comment by Birkemose** [comment]
The problem was, that previously setting interval to 0, caused the selector to be called each frame. Now it apparently gets called all the time, and stops the app execution.
Also
In general, rescheduling a selector, doesn't produce any warnings. Not sure if it should.

**Comment by slembcke** [comment]
Ah, yeah. I figured out that it would go into an infinite loop with an interval of 0.0 from #519.

Since the new scheduler has sub-frame accuracy to support fixed time-stepping, if you asked it to use an interval of 0.0 on an infinitely repeating timer it would never be able to step past it.

I added assertions to warn about attempting to explicitly schedule the update: or fixedUpdate: methods as well as an assertion to catch repeating timers with an interval of 0.0. Should be pushed to Git now.

Did Cocos 2.x warn about rescheduling selectors? It could be added easily enough. It sort of does make sense.
